terraform/environments/data-platform/glue_job.tf

```diff
@@ -0,0 +1,122 @@
+
+locals {
+  default_arguments = {
+    "--continuous-log-logGroup" = aws_cloudwatch_log_group.log_group.name
+  }
+}
+
+
+# load the glue script to bucket as bucket object
+resource "aws_s3_object" "object" {
+  bucket=module.s3-bucket.bucket.id
+  key    = "glue_script/glue_transform_script.py"
+  source = "glue_script/glue_transform_script.py"
+  etag = filemd5("glue_script/glue_transform_script.py")
+}
+
+resource "aws_glue_job" "glue_job" {
+  name        = "${local.name}-glue-job-${local.environment}"
+  role_arn    = aws_iam_role.glue-service-role.arn
+  description            = "Glue job to transform and load data products to the glue catalog"
+  glue_version           = local.glue_version
+  max_retries            = local.max_retries
+  security_configuration = aws_glue_security_configuration.sec_cfg.id
+  worker_type            = local.worker_type
+  number_of_workers      = local.number_of_workers
+  timeout                = local.timeout
+  execution_class        = local.execution_class
+  tags                   = local.tags
+
+  command {
+    script_location = "s3://${module.s3-bucket.bucket.id}/glue_script/glue_transform_script.py"
+    name            = "glueetl"
+  }
+
+  default_arguments = merge(local.glue_default_arguments, local.default_arguments)
+
+  execution_property {
+    max_concurrent_runs = local.max_concurrent
```

Comment:
Can this be passed from Lambda which kick starts the Glue. This gives us the flexibility to increase depending on the work load . Max concurrent is 100 per region i think

Reply:
I don't think this is possible as needs to be defined in the infra and is not a parameter that can be passed like some of the other things passed in as default arguments or from the calling lambda, see [job parameters](https://docs.aws.amazon.com/glue/latest/dg/aws-glue-programming-etl-glue-arguments.html).
I put 5 as a holding position in locals but not sure what the actual figure should be

Reply:
If you anticipate having a scenario where 100 concurrent runs are needed, why not just set that value to 100? It won't kick off that many concurrent executions unless it needs to do so.

Reply:
This might be different for each account . Say sandbox might be different than production systems. So may be we can set some default and pass on in lambda when we start the job

Reply:
Outside the scope of this PR for sure, but a way we've managed this before is to set up a local for each environment that's then gets referenced depending on what workspace the terraform is running in. It's quite cool!

Reply:
oh thats nice to know. Will look into it . If you can share the link of the github that will be useful